app/uk/gov/hmrc/incorporatedentityidentificationfrontend/connectors/CreateBusinessVerificationJourneyConnector.scala

```diff
@@ -57,18 +58,20 @@ class CreateBusinessVerificationJourneyConnector @Inject()(http: HttpClient,
 }
 
 object CreateBusinessVerificationJourneyConnector {
-  implicit object BusinessVerificationHttpReads extends HttpReads[Option[String]] {
-    override def read(method: String, url: String, response: HttpResponse): Option[String] = {
+  implicit object BusinessVerificationHttpReads extends HttpReads[(Option[String],Option[BusinessVerificationStatus])] {
```

Comment:
This might be better represented as an algebraic data type, such as:

Either[JourneyCreationFailure, JourneyCreated]
case class JourneyCreated(redirectUri: String)
sealed trait JourneyCreationFailure
case object NotEnoughEvidence extends JourneyCreationFailure
case object UserLockedOut extends JourneyCreationFailure

You might need a service layer to translate this into something the controller understands, but you could have a service class that calls this API, stores the result of it (if it has failed) and returns the redirect URL (if it's successful), then the controller would just need to redirect if there is an available redirect, or go elsewhere if not.

The service layer might not be needed at this point. Probably worth just seeing how much code it is.

Reply:
I've done the refactor